martianurl/url_count_verifier.go

```diff
@@ -0,0 +1,173 @@
+package martianurl
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/url"
+	"regexp"
+	"sync"
+
+	"github.com/google/martian/parse"
+)
+
+func init() {
+	parse.Register("urlcount.Verifier", urlCountVerifierFromJSON)
+}
+
+type urlCountVerifier struct {
+	urlPattern *regexp.Regexp
+	params     map[string]*regexp.Regexp
+	subParams  map[string]map[string]*regexp.Regexp
+	count      int
+
+	mu      sync.Mutex
+	counted int
+}
+
+type urlCountVerifierJSON struct {
+	Count      int                    `json:"count"`
+	URLPattern string                 `json:"urlPattern"`
+	Params     map[string]interface{} `json:"params"`
+	Scope      []parse.ModifierType   `json:"scope"`
+}
+
+func NewURLCountVerifier(urlpattern string, count int) (*urlCountVerifier, error) {
```

Comment:
I'd rather this take a *regexp.Regexp instead of a string - force the compilation of the regular expression as early as possible.